### PR TITLE
[PROTOCOL-336] Update getTotalOwed function

### PR DESCRIPTION
**Pull request by sypeer**
It is:
- Updated getTotalOwed function
- Updated _getCollateralNeededInTokens
 
Screen Shot 2020-11-04 at 5 35 07 PM

### CODE REVIEW

File: contracts/base/LoansBase.sol [hunk 1]
```diff
@@ -568,18 +568,31 @@ contract LoansBase is LoansInterface, Base {
...
+     function _getTotalOwed(uint256 loanID)
+        internal
+        view
+        returns (uint256)
     {
         TellerCommon.LoanStatus currentStatus = loans[loanID].status;
         uint256 loanAmount;
         if (currentStatus == TellerCommon.LoanStatus.TermsSet) {
-            loanAmount = loans[loanID].loanTerms.maxLoanAmount;
+            loanAmount = loans[loanID].loanTerms.maxLoanAmount.add(loans[loanID].interestOwed);
```
Comment (passabilities) [inline contracts/base/LoansBase.sol:589]: Interest owed won't be set yet if the loan is not yet active. So we'll have to calculate the interest based on the max loan amount. 

Take the calculation code for the interest that is in `takeOutLoan` here:
https://github.com/teller-protocol/teller-protocol-v1/blob/72a6d1d17f77a1226a5f3a1855ae1fec2e02da04/contracts/base/LoansBase.sol#L254-L258
and maybe put it in its own internal function to reuse.

**Review by passabilities — CHANGES_REQUESTED** [review]
Also write a simple unit test/integration test for this to check the values being calculated.

File: contracts/base/LoansBase.sol (hunk 1 above)
Comment (sypeer) [reply contracts/base/LoansBase.sol]: Changed accordingly

File: contracts/base/LoansBase.sol [hunk 2]
```diff
@@ -586,7 +582,8 @@ contract LoansBase is LoansInterface, Base {
         TellerCommon.LoanStatus currentStatus = loans[loanID].status;
         uint256 loanAmount;
         if (currentStatus == TellerCommon.LoanStatus.TermsSet) {
-            loanAmount = loans[loanID].loanTerms.maxLoanAmount.add(loans[loanID].interestOwed);
+            uint256 interestOwed = _getInterestOwed(loanID, loans[loanID].loanTerms.maxLoanAmount);
+            loanAmount = loans[loanID].loanTerms.maxLoanAmount.add(interestOwed);
         } else if (currentStatus == TellerCommon.LoanStatus.Active) {
             loanAmount = loans[loanID].interestOwed.add(loans[loanID].principalOwed);
```
Comment (passabilities) [inline contracts/base/LoansBase.sol:588]: Just a preference to have the principalOwed before the interest 😅

File: contracts/base/LoansBase.sol [hunk 3]
```diff
@@ -595,6 +592,24 @@ contract LoansBase is LoansInterface, Base {
...
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 interestOwed = amountBorrow
+            .mul(loans[loanID].loanTerms.interestRate)
+            .mul(loans[loanID].loanTerms.duration)
+            .div(TEN_THOUSAND)
+            .div(SECONDS_PER_YEAR_4DP);
+        return interestOwed;
```
Comment (passabilities) [inline contracts/base/LoansBase.sol:610]: lets just return the value instead of creating a variable

**Review by passabilities — CHANGES_REQUESTED** [review]

File: contracts/base/LoansBase.sol [hunk 4]
```diff
@@ -568,18 +564,50 @@ contract LoansBase is LoansInterface, Base {
...
         if (currentStatus == TellerCommon.LoanStatus.TermsSet) {
-            loanAmount = loans[loanID].loanTerms.maxLoanAmount;
+            uint256 interestOwed = _getInterestOwed(loanID, loans[loanID].loanTerms.maxLoanAmount);
+            loanAmount = loans[loanID].loanTerms.maxLoanAmount.add(interestOwed);
         } else if (currentStatus == TellerCommon.LoanStatus.Active) {
-            loanAmount = getTotalOwed(loanID);
+            loanAmount = loans[loanID].interestOwed.add(loans[loanID].principalOwed);
         } else {
-            return 0;
+            loanAmount = 0;
```
Comment (passabilities) [inline contracts/base/LoansBase.sol:590]: also here you can just return 0. no need to set the variable

File: contracts/base/LoansBase.sol (hunk 2 above)
Comment (sypeer) [reply contracts/base/LoansBase.sol]: Got it, changed

File: contracts/base/LoansBase.sol (hunk 3 above)
Comment (sypeer) [reply contracts/base/LoansBase.sol]: Done

File: contracts/base/LoansBase.sol (hunk 4 above)
Comment (sypeer) [reply contracts/base/LoansBase.sol]: Done

**Review by passabilities — APPROVED** [review]

**Review by salazarguille — APPROVED** [review]

**Pull request merged**